includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -33,15 +40,17 @@ class WC_Stripe_Apple_Pay_Registration {
 
 	public function __construct() {
 		add_action( 'init', array( $this, 'add_domain_association_rewrite_rule' ) );
+		add_action( 'admin_init', [ $this, 'verify_domain_on_domain_name_change' ] );
+		add_action( 'admin_notices', array( $this, 'admin_notices' ) );
 		add_filter( 'query_vars', array( $this, 'whitelist_domain_association_query_param' ), 10, 1 );
 		add_action( 'parse_request', array( $this, 'parse_domain_association_request' ), 10, 1 );
 
 		add_action( 'woocommerce_stripe_updated', array( $this, 'verify_domain_if_configured' ) );
 		add_action( 'add_option_woocommerce_stripe_settings', array( $this, 'verify_domain_on_new_settings' ), 10, 2 );
 		add_action( 'update_option_woocommerce_stripe_settings', array( $this, 'verify_domain_on_updated_settings' ), 10, 2 );
-		add_action( 'admin_notices', array( $this, 'admin_notices' ) );
 
 		$this->stripe_settings         = get_option( 'woocommerce_stripe_settings', array() );
+		$this->domain_name             = isset( $_SERVER['HTTP_HOST'] ) ? $_SERVER['HTTP_HOST'] : str_replace( [ 'https://', 'http://' ], '', get_site_url() ); // @codingStandardsIgnoreLine
```

Comment:
I think this is a good case for the [null coalescing operator](https://www.php.net/manual/en/language.operators.comparison.php#language.operators.comparison.coalesce) 🙂 

```suggestion
		$this->domain_name             = $_SERVER['HTTP_HOST'] ?? str_replace( [ 'https://', 'http://' ], '', get_site_url() ); // @codingStandardsIgnoreLine
```

Reply:
We could use it, but apparently this operator is a relatively new thing and was introduced in PHP 7.0. We still need to support PHP 5.2, so I believe we shouldn't use it.

https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op

Reply:
Ah, of course. I'm used to being able to use these newer features in WooCommerce Payments. Just ignore this! 😛